tests/test_interface.rb

```diff
@@ -141,15 +141,15 @@ def interface_count
   def capable_speed_values(interface)
     capabilities = config("show interface #{interface.name} capabilities")
     speed_capa = capabilities.match(/Speed:\s+(\S+)/)
-    return [] if speed_capa[1].nil?
+    return [] if speed_capa.nil? || speed_capa[1].nil?
```

Comment:
I think it might be helpful to log a warning when speed and duplex capabilities cannot be detected.